src/pyhf/workspace.py

```diff
@@ -3,33 +3,41 @@
 from . import exceptions
 from . import utils
 from .pdf import Model
+from .mixins import _ChannelSummaryMixin
 
 logging.basicConfig()
 log = logging.getLogger(__name__)
 
 
-class Workspace(object):
+class Workspace(_ChannelSummaryMixin, dict):
     """
-    An object that is built from a JSON spec that follows `workspace.json`.
+    A JSON-serializable object that is built from an object that follows the `workspace.json` schema.
     """
 
     def __init__(self, spec, **config_kwargs):
-        self.spec = spec
-
+        super(Workspace, self).__init__(spec, channels=spec['channels'])
         self.schema = config_kwargs.pop('schema', 'workspace.json')
         self.version = config_kwargs.pop('version', None)
         # run jsonschema validation of input specification against the (provided) schema
         log.info("Validating spec against schema: {0:s}".format(self.schema))
-        utils.validate(self.spec, self.schema, version=self.version)
+        utils.validate(self, self.schema, version=self.version)
 
         self.measurement_names = []
-        for measurement in self.spec.get('measurements', []):
+        for measurement in self.get('measurements', []):
             self.measurement_names.append(measurement['name'])
 
         self.observations = {}
-        for obs in self.spec['observations']:
+        for obs in self['observations']:
             self.observations[obs['name']] = obs['data']
 
+    def __eq__(self, other):
```

Comment:
do we eally want this high level logic?

Reply:
LGTM complains if I don't include it...

Reply:
See [the __eq__](https://lgtm.com/projects/g/scikit-hep/pyhf/rev/pr-710232ea6f84678b0578b28d44820bfb7ff5cbf1) warning and the subsequent [__ne__](https://lgtm.com/projects/g/scikit-hep/pyhf/rev/pr-033b34773cbd178d22e6e5a4308022eb37878f5c) warning.

Reply:
ok